lib/project_types/extension/models/specification_handlers/beacon_extension.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+require "base64"
+require "fileutils"
+require "json"
+require_relative "beacon_extension_utils/script_config"
+require_relative "beacon_extension_utils/script_config_repository"
+
+module Extension
+  module Models
+    module SpecificationHandlers
+      class BeaconExtension < Default
+        SCRIPT_FILE = "src/index.js"
+
+        def name
+          "Beacon Extension"
+        end
+
+        def read_configuration
+        end
+
+        def read_file_contents(context, file_path, &read_file_block)
+          read_file_block.call(file_path)
+        rescue StandardError
+          context.abort(context.message("core.extension.push.beacon_extension.error.file_read_error", file_path))
+        end
+
+        def access_config_property(context, ext_config, key, &process_value)
+          context.abort(context.message("core.extension.push.beacon_extension.error.missing_config_key_error",
+            key)) unless ext_config.key?(key)
+
+          begin
+            process_value.nil? ? ext_config[key] : process_value.call(ext_config[key])
+          rescue StandardError
+            context.abort(context.message("core.extension.push.beacon_extension.error.invalid_config_value_error", key))
+          end
+        end
+
+        def config(context)
+          begin
+            ext_config = BeaconExtensionUtils::ScriptConfigYmlRepository.new(ctx: context).get!.content
+          rescue StandardError
+            context.abort(context.message("core.extension.push.beacon_extension.error.file_read_error",
+              "extension.config.yml"))
```

Comment:
should we jsut reference `ScriptConfigYmlRepository.filename` here?

Reply:
fixed